src/utils/mod.rs

```diff
@@ -236,18 +240,31 @@ pub fn find_big_ones(new_l: Vec<(String, u64)>, max_to_show: usize) -> Vec<(Stri
 }
 
 pub fn trim_deep_ones(
-    input: Vec<(String, u64)>,
+    input: Vec<(PathBuf, u64)>,
     max_depth: u64,
-    top_level_names: &HashSet<String>,
-) -> Vec<(String, u64)> {
-    let mut result: Vec<(String, u64)> = Vec::with_capacity(input.len() * top_level_names.len());
+    top_level_names: &HashSet<PathBuf>,
+) -> Vec<(PathBuf, u64)> {
+    let mut result: Vec<(PathBuf, u64)> = Vec::with_capacity(input.len() * top_level_names.len());
 
     for name in top_level_names {
-        let my_max_depth = name.matches(std::path::is_separator).count() + max_depth as usize;
-        let name_ref: &str = name.as_ref();
+        let my_max_depth = name
+            .components()
+            .filter(|&c| match c {
+                std::path::Component::Prefix(_) => false,
+                _ => true,
+            })
```

Comment:
can't we remove the above `filter` isn't name.components.count() enough?

Reply:
No, paths can have some odd preliminary ("Prefix") bits (for example, from windows, "\\?\" or "\\UNC\") which need to be removed here to calculate the depth correctly.